biz.aQute.bndlib/src/aQute/bnd/differ/JavaElement.java

```diff
@@ -685,26 +685,28 @@ private String toString(TypeRef[] prototype) {
 		return sb.toString();
 	}
 
-	static Element	BOOLEAN_R	= new Element(RETURN, "boolean");
-	static Element	BYTE_R		= new Element(RETURN, "byte");
-	static Element	SHORT_R		= new Element(RETURN, "short");
-	static Element	CHAR_R		= new Element(RETURN, "char");
-	static Element	INT_R		= new Element(RETURN, "int");
-	static Element	LONG_R		= new Element(RETURN, "long");
-	static Element	FLOAT_R		= new Element(RETURN, "float");
-	static Element	DOUBLE_R	= new Element(RETURN, "double");
+	static final Element	VOID_R		= new Element(RETURN, "void");
+	static final Element	BOOLEAN_R	= new Element(RETURN, "boolean");
+	static final Element	BYTE_R		= new Element(RETURN, "byte");
+	static final Element	SHORT_R		= new Element(RETURN, "short");
+	static final Element	CHAR_R		= new Element(RETURN, "char");
+	static final Element	INT_R		= new Element(RETURN, "int");
+	static final Element	LONG_R		= new Element(RETURN, "long");
+	static final Element	FLOAT_R		= new Element(RETURN, "float");
+	static final Element	DOUBLE_R	= new Element(RETURN, "double");
+	static final Element	OBJECT_R	= new Element(RETURN, "java.lang.Object");
 
 	private void getCovariantReturns(Collection<Element> elements, TypeRef type, String generics) throws Exception {
-		if (type == null || type.isObject())
```

Comment:
Discusses with Peter. We do need to add Object and void as return types to detect changes to the return types from void to non-void and from Object to another type. Without added Object and void as return types, a change between void and Object is unnoticed and a change from void or Object to some other return type is only seen as an add only. The remove of void or Object is unnoticed.